lib/internal/util.js

```diff
@@ -179,3 +180,18 @@ exports.toLength = function toLength(argument) {
   const len = toInteger(argument);
   return len <= 0 ? 0 : Math.min(len, Number.MAX_SAFE_INTEGER);
 };
+
+const OS_SIGNALS_TO_NAMES_MAPPING = Object.create(null);
+Object.keys(OS_SIGNALS).forEach((signal) => {
+  OS_SIGNALS_TO_NAMES_MAPPING[OS_SIGNALS[signal]] = signal;
+});
+
+exports.convertToValidSignal = function convertToValidSignal(signal) {
+  if (typeof signal === 'number' && signal in OS_SIGNALS_TO_NAMES_MAPPING)
+    return signal;
+
+  if (typeof signal === 'string' && OS_SIGNALS.hasOwnProperty(signal))
+    return OS_SIGNALS[signal];
+
+  throw new Error('Unknown signal: ' + signal);
+};
```

Comment:
We should avoid `in`, `hasOwnProperty()`, and unnecessarily checking the type again when the secondary condition fails. How about something like this:

```js
exports.convertToValidSignal = function convertToValidSignal(signal) {
  if (typeof signal === 'number') {
    if (OS_SIGNALS_TO_NAMES_MAPPING[signal])
      return signal;
  } else if (typeof signal === 'string') {
    var val = OS_SIGNALS[signal];
    if (val !== undefined)
      return val;
  }
  throw new Error('Unknown signal: ' + signal);
};
```